src/transactions/financial.ts

```diff
@@ -263,3 +309,61 @@ export const constructTransferCreatorCoinTransaction = (
     }
   );
 };
+
+const buildBuyCreatorCoinMetadata = (params: BuyCreatorCoinRequestParams) => {
+  const metadata = new TransactionMetadataCreatorCoin();
+  metadata.profilePublicKey = bs58PublicKeyToCompressedBytes(
+    params.CreatorPublicKeyBase58Check
+  );
+  metadata.operationType = 0;
+  metadata.desoToSellNanos = params.DeSoToSellNanos;
+  metadata.desoToAddNanos = params.DeSoToAddNanos ?? 0;
+  metadata.minCreatorCoinExpectedNanos = params.MinDeSoExpectedNanos ?? 0;
+  metadata.minDeSoExpectedNanos = params.MinDeSoExpectedNanos ?? 0;
+
+  return metadata;
+};
+
+const buildSellCreatorCoinMetadata = (params: SellCreatorCoinRequestParams) => {
+  // NOTE: I'm not sure if this is completely correct... I'm not sure if it matters
+  // for just checking permissions. If we are using this for local tx construction
+  // it might need to be updated.
+  const metadata = new TransactionMetadataCreatorCoin();
+  metadata.profilePublicKey = bs58PublicKeyToCompressedBytes(
+    params.CreatorPublicKeyBase58Check
+  );
+  metadata.operationType = 1;
+  metadata.creatorCoinToSellNanos = params.CreatorCoinToSellNanos;
+  metadata.desoToAddNanos = params.DeSoToAddNanos ?? 0;
+  metadata.minCreatorCoinExpectedNanos = params.MinDeSoExpectedNanos ?? 0;
+  metadata.minDeSoExpectedNanos = params.MinDeSoExpectedNanos ?? 0;
+
+  return metadata;
+};
+
+// TODO: Make sure these are good to use for full local tx construction.
```

Comment:
I would still block devs from using this function for local construction since we don't have the math for CC curves to properly compute MinDeSoExpectedNanos and MinCreatorCoinExpectedNanos

Reply:
cool I just deleted them since they were unused and never exported anyway.